lib/core/public/load.js

```diff
@@ -18,7 +18,10 @@ function runCommand(data, keepalive, callback) {
 
 	switch (data.command) {
 		case 'rules':
-			return runRules(context, options, resolve, reject);
+			return runRules(context, options, function (results, cleanup) {
+				resolve(results);
+				cleanup();
```

Comment:
Should `cleanup` be called before we `resolve()` here? I don't know if `resolve` is synchronous or not.

Reply:
It's synchronous, and it generates selectors so it needs to go before we clean up the selectorData.

Reply:
Cool, OK. Thank you for clarifying.